### PR TITLE
Enable search by validator name

### PR DESCRIPTION
**Pull request by buberdds**
branch
https://pr-1633.oasis-explorer.pages.dev/mainnet/consensus/search?q=dust
master
https://explorer.dev.oasis.io/mainnet/consensus/search?q=dust

### CODE REVIEW

File: src/app/hooks/useSearchForValidatorByName.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  return matchedAddresses
+}
+
+export const useSearchForValidatorByName = (
+  network: Network,
+  nameFragment: string | undefined,
+): AccountNameSearchResults => {
+  const { isLoading, isError, error, data } = useGetConsensusValidatorsAddressNameMap(network)
+  if (isError) {
+    console.log('Failed to load Validator data', error)
```
Comment (lubej) [inline src/app/hooks/useSearchForValidatorByName.ts:29]: Do we want `console.log` here?

File: src/app/pages/SearchResultsPage/hooks.ts [hunk 2]
```diff
@@ -234,6 +235,21 @@ export function useNamedAccountConditionally(
   }
 }
 
+export function useNamedValidatorConditionally(nameFragment: string | undefined) {
+  const queries = RouteUtils.getEnabledNetworksForLayer(Layer.consensus).map(network =>
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useSearchForValidatorByName(network, nameFragment),
```
Comment (lubej) [inline src/app/pages/SearchResultsPage/hooks.ts:241]: Don't particularly like this, it should be something like `useSearchForValidatorByNames`, which would receive an array of networks.

File: src/app/components/Validators/ValidatorLink.tsx [hunk 3]
```diff
@@ -15,15 +15,15 @@ type ValidatorLinkProps = {
...
+  highlightedPartOfName?: string
 }
 
 export const ValidatorLink: FC<ValidatorLinkProps> = ({
   address,
   name,
   network,
   alwaysTrim,
-  highlightedPart,
+  highlightedPartOfName,
```
Comment (lubej) [inline src/app/components/Validators/ValidatorLink.tsx:26]: Just a nit. But naming doesn't seems consistent, I would keep `highlightedPart`.

**Review by lubej — APPROVED** [review]

File: src/app/hooks/useSearchForValidatorByName.ts (hunk 1 above)
Comment (buberdds) [reply src/app/hooks/useSearchForValidatorByName.ts:29]: removed

File: src/app/pages/SearchResultsPage/hooks.ts (hunk 2 above)
Comment (buberdds) [reply src/app/pages/SearchResultsPage/hooks.ts:241]: Name updated, but I will keep that pattern where we create a hook just to brake a rule of hooks. This is how entire search was built. Will create an issue to refactor all at once.

Comment (buberdds) [reply src/app/pages/SearchResultsPage/hooks.ts]: https://github.com/oasisprotocol/explorer/issues/1635

**Pull request merged**